tyrian/js/src/main/scala/tyrian/TyrianAppF.scala

```diff
@@ -96,3 +99,47 @@ trait TyrianAppF[F[_]: Async, Msg, Model]:
 
       case None =>
         throw new Exception(s"Missing Element! Could not find an element with id '$containerId' on the page.")
+
+object TyrianAppF:
+  /** Launch app instances after DOMContentLoaded.
+    */
+  def launchOnContentLoaded[F[_] : Async](appDirectory: Map[String, () => TyrianAppF[F, _, _]]): Unit = {
+    val documentReady = new js.Promise((resolve, _reject) => {
```

Comment:
If the main `<script>` tag uses `module` or `defer` then it looks like the DOM will be ready when the script is loaded. In that case you could call `launch` directly.

```
<script type="module" src="/index.hash1.js"></script>
<script src="/index.hash2.js" nomodule defer></script>
```